### PR TITLE
chore: Speed up CI even more

### PR DESCRIPTION
**Pull request by NicholasLYang**
### Description

Speeds up CI by using sccache along with turbo

### Testing Instructions

Closes TURBO-1420

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -224,12 +224,26 @@ jobs:
...
           github-token: "${{ secrets.GITHUB_TOKEN }}"
 
       - run: turbo run build --filter=cli --color --env-mode=strict --token=${{ secrets.TURBO_TOKEN }} --team=${{ vars.TURBO_TEAM }}
+        env:
+          SCCACHE_BUCKET: turborepo-sccache
+          SCCACHE_REGION: us-east-2
+          RUSTC_WRAPPER: "sccache"
+          CARGO_INCREMENTAL: 0
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```
Comment (mehulkar) [inline .github/workflows/test.yml:242]: Does cargo automatically pick these up?

File: crates/turborepo-lib/src/cli.rs [hunk 2]
```diff
@@ -204,11 +204,10 @@ impl Args {
     pub fn new() -> Result<Self> {
         // We always pass --single-package in from the shim.
         // We need to omit it, and then add it in for run.
-        let arg_separator_position =
-            std::env::args_os().position(|input_token| input_token == "--");
+        let arg_separator_position = env::args_os().position(|input_token| input_token == "--");
 
         let single_package_position =
-            std::env::args_os().position(|input_token| input_token == "--single-package");
+            env::args_os().position(|input_token| input_token == "--single-package");
```
Comment (mehulkar) [inline crates/turborepo-lib/src/cli.rs:210]: Why are these changes in this PR? If this is just lint fixes, can you make them separately?

**Review by mehulkar — APPROVED** [review]
🚢 🔧 . Approving, but would prefer to get unrelated changes out of this commit

File: crates/turborepo-lib/src/cli.rs (hunk 2 above)
Comment (NicholasLYang) [reply crates/turborepo-lib/src/cli.rs:210]: Yeah this was just to test the CI. I'll revert them out

File: .github/workflows/test.yml (hunk 1 above)
Comment (NicholasLYang) [reply .github/workflows/test.yml:242]: sccache does, yeah

**Pull request merged**